### ISSUE TITLE
Missing networking related kernel modules for v1.6.8-rc1 on RHEL 7.4 hosts with native docker and SElinux enabled.

### ISSUE THREAD

**Issue by moelsayed** [rancher/rancher#9676]
**Rancher versions:**
rancher/server: v1.6.8-rc1

**Docker version: (`docker version`,`docker info` preferred)**
1.12 native
**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
RHEL-4
**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)*
AWS
**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node
**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
cattle/k8s
**Steps to Reproduce:**
* Install v1.6.8-rc1 on RHEL 7.4 with native docker 1.12
**Results:**
K8s(if used), healthcheck and scheduler fail to start. ipsec containers can't ping each other.

**Comment by leodotcloud** [comment]
@moelsayed Can you please share steps to install native docker on the RHEL servers?

**Comment by leodotcloud** [comment]
Used the following steps to install native docker and reproduce the issue:
```
sudo yum-config-manager --enable rhui-REGION-rhel-server-extras
sudo yum -y install docker
```

**Comment by leodotcloud** [comment]
I see the following messages in `/var/log/audit/audit.log`:
```
type=AVC msg=audit(1503074920.707:3806): avc:  denied  { module_request } for  pid=31667 comm="ip" kmod="crypto-rfc4106-gcm-aesni" scontext=system_u:system_r:svirt_lxc_net_t:s0:c342,c844 tcontext=system_u:system_r:kernel_t:s0 tclass=system
type=AVC msg=audit(1503074920.707:3806): avc:  denied  { module_request } for  pid=31667 comm="ip" kmod="crypto-rfc4106-gcm-aesni-all" scontext=system_u:system_r:svirt_lxc_net_t:s0:c342,c844 tcontext=system_u:system_r:kernel_t:s0 tclass=system
type=AVC msg=audit(1503074933.575:3825): avc:  denied  { module_request } for  pid=31706 comm="charon" kmod="xfrm-mode-10-1" scontext=system_u:system_r:svirt_lxc_net_t:s0:c342,c844 tcontext=system_u:system_r:kernel_t:s0 tclass=system
```

**Comment by leodotcloud** [comment]
We are supposed to be loading a bunch of kernel modules from the `rancher-agent` container: https://github.com/rancher/rancher/blob/v1.6.8-rc4/agent/run.sh#L35
But on the RHEL7.4 host, I don't find these modules loaded.

```
-bash-4.2# lsmod | grep "ansi_cprng \|drbg \|esp4 \|veth \|xfrm4_mode_tunnel \|xfrm6_mode_tunnel \|xt_mark \|xt_nat"
xt_nat                 12681  6
xt_mark                12563  4
veth                   13410  0
nf_nat                 26787  3 nf_nat_ipv4,xt_nat,nf_nat_masquerade_ipv4
-bash-4.2#
```

**Comment by leodotcloud** [comment]
Workaround for now: Load the modules from the host.
```
modprobe ansi_cprng
modprobe drbg
modprobe esp4
modprobe xfrm4_mode_tunnel
modprobe xfrm6_mode_tunnel
modprobe vxlan
```

**Comment by leodotcloud** [comment]
On a host with selinux disabled, manually starting the agent container with `--entrypoint=bash` and trying to load the modules still has the same issue.
```
-bash-4.2# sestatus
SELinux status:                 disabled
-bash-4.2# docker exec -it 873 bash
root@873742f63f4c:/# modprobe ah4
modprobe: ERROR: could not insert 'ah4': Exec format error
root@873742f63f4c:/#
```

**Comment by leodotcloud** [comment]
Host:
```
-bash-4.2# modprobe --version
kmod version 20
-bash-4.2#
```

Container:
```
root@ip-172-31-59-252:/# modprobe --version
kmod version 22
-XZ -ZLIB -EXPERIMENTAL
```

After copying the kmod binary from the host to container, I am able to load the kernel modules from the container. Looks like the version mismatch is causing problems loading the kernel modules.

**Comment by deniseschannon** [comment]
Available in v1.6.8-rc6 with rancher/agent:v1.2.6-rc3

**Comment by moelsayed** [comment]
Tested with the new rc. catller validation passed.
